pkg/controller/controller_utils.go

```diff
@@ -290,6 +293,39 @@ func (r RealPodControl) CreateReplica(namespace string, controller *api.Replicat
 	return nil
 }
 
+func (r RealPodControl) CreateReplicaOnNode(namespace string, controller *expapi.Daemon, nodeName string) error {
+	desiredLabels := getReplicaLabelSet(controller.Spec.Template)
+	desiredAnnotations, err := getReplicaAnnotationSet(controller.Spec.Template, controller)
+	if err != nil {
+		return err
+	}
+	prefix := getReplicaPrefix(controller.Name)
+
+	pod := &api.Pod{
+		ObjectMeta: api.ObjectMeta{
+			Labels:       desiredLabels,
+			Annotations:  desiredAnnotations,
+			GenerateName: prefix,
+		},
+	}
+	if err := api.Scheme.Convert(&controller.Spec.Template.Spec, &pod.Spec); err != nil {
+		return fmt.Errorf("unable to convert pod template: %v", err)
+	}
+	if labels.Set(pod.Labels).AsSelector().Empty() {
```

Comment:
Cool, maybe we can add a comment "If pod does not have labels then it cannot be controlled by any daemon controller"